src/DeviceInput/InputDevices/deviceEnums.ts

```diff
@@ -0,0 +1,199 @@
+/**
+ * Enum for Device Types
+ */
+export enum DeviceType {
+    /** Keyboard */
+    Keyboard = 0,
+    /** Mouse */
+    Mouse = 1,
+    /** Touch Pointers */
+    Touch = 2,
+    /** PS4 Dual Shock */
+    DualShock = 3,
+    /** Xbox */
+    Xbox = 4,
+    /** Switch Controller */
+    Switch = 5,
+    /** Generic */
+    Generic = 6

```

Comment:
If we are going to keep `Generic`, can we set its value to `Number.MAX_VALUE` or something so that when we add additional device types in the future, `Generic` can remain at the very end of the enum?

Reply:
With the way that we've implemented keeping track of the devices, I feel like setting this to max wouldn't be the best course of action.  We use the number of enums in DeviceType to create and allocate an array and then copy the data into their corresponding areas using the associated enum's assigned number.

Reply:
Maybe make Generic = 0 and shift everything else?

Reply:
I'd be fine with that.